bot/exts/events/advent_of_code/_cog.py

```diff
@@ -172,6 +175,72 @@ async def join_leaderboard(self, ctx: commands.Context) -> None:
+    async def aoc_link_account(self, ctx: commands.Context, aoc_name: str = None) -> None:
+        """
+        Link your Discord Account to your Advent of Code name.
+
+        Stored in a Redis Cache, Discord ID: Advent of Code Name
+        """
+        cache_items = await self.account_links.items()
+
+        # A short circuit in case the cache is empty
+        if len(cache_items) == 0 and aoc_name:
+            log.info(f"{ctx.author} ({ctx.author.id}) is now linked to {aoc_name}.")
+            await self.account_links.set(ctx.author.id, aoc_name)
+            await ctx.reply(f"You have linked your Discord ID to {aoc_name}.")
+            return
+        elif len(cache_items) == 0:
+            await ctx.reply(
+                "You have not linked an Advent of Code account."
+                "Please re-run the command with one specified."
+            )
+            return
+
+        cache_aoc_name = [value for _, value in cache_items]
+
+        if aoc_name:
+            # Let's check the current values in the cache to make sure it isn't already tied to a different account
+            if aoc_name == await self.account_links.get(ctx.author.id):
+                await ctx.reply(f"{aoc_name} is already tied to your account.")
+                return
+            elif aoc_name in cache_aoc_name:
+                log.info(
+                    f"{ctx.author} ({ctx.author.id}) tried to connect their account to {aoc_name},"
+                    " but it's already connected to another user."
+                )
+                await ctx.reply(
+                    f"{aoc_name} is already tied to another account."
+                    " Please contact an admin if you believe this is an error."
+                )
+                return
+
+            # Update an existing link
+            if old_aoc_name := await self.account_links.get(ctx.author.id):
+                log.info(f"{ctx.author} ({ctx.author.id}) has changed their link from {old_aoc_name} to {aoc_name}.")
+                await self.account_links.set(ctx.author.id, aoc_name)
+                await ctx.reply(f"Your linked account has been changed to {aoc_name}.")
+            else:
+                # Create a new link
+                log.info(f"{ctx.author} ({ctx.author.id}) is now linked to {aoc_name}.")
+                await self.account_links.set(ctx.author.id, aoc_name)
+                await ctx.reply(f"You have linked your Discord ID to {aoc_name}.")
+        else:
+            # User has not supplied a name, let's check if they're in the cache or not
+            if cache_name := await self.account_links.get(ctx.author.id):
+                await ctx.reply(f"You have already linked an Advent of Code account: {cache_name}.")
+            else:
+                await ctx.reply(
+                    "You have not linked an Advent of Code account."
+                    " Please re-run the command with one specified."
+                )
```

Comment:
This could probably be the first part of the if statement with `if not aoc_name` and then we could return early to avoid one extra indentation 🙃

Reply:
No comment on this and it's not mentioned, yet it is resolved. 

Looking at the code again, it wouldn't really create much of a difference so could be skipped.